CMakeLists.txt

```diff
@@ -710,15 +710,14 @@ if(UBSAN)
     message(FATAL_ERROR "Cannot enable UBSAN unless using Clang")
   endif()
 
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsanitize=undefined")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=undefined")
-  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=undefined")
-
+  set(SANITIZER_FLAGS "-fsanitize=undefined,unsigned-integer-overflow")
```

Comment:
Should add a comment about why `unsigned-integer-overflow` is listed (and not other options)? We might also reference the [clang documentation](https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html#id6).  

I was going to ask why `signed-integer-overflow` was not included then saw that it's covered by `-fsanitize=undefined`:


> `-fsanitize=undefined`: All of the checks listed above other than `float-divide-by-zero`, `unsigned-integer-overflow`, `implicit-conversion`, `local-bounds` and the `nullability-*` group of checks.